niworkflows/interfaces/cifti.py

```diff
@@ -68,112 +78,141 @@ class GenerateCifti(SimpleInterface):
+        return variants[space]
 
     def _fetch_data(self):
-        """Convert inputspec to files."""
-        if (self.inputs.surface_target == "fsnative" or
-                self.inputs.volume_target != "MNI152NLin2009cAsym"):
-            # subject space is not support yet
-            raise NotImplementedError
-
-        annotation_files = sorted(glob(os.path.join(self.inputs.subjects_dir,
-                                                    self.inputs.surface_target,
-                                                    'label',
-                                                    '*h.aparc.annot')))
-        if not annotation_files:
-            raise IOError("Freesurfer annotations for %s not found in %s" % (
-                          self.inputs.surface_target, self.inputs.subjects_dir))
-
-        label_file = str(get_template(
-            'MNI152NLin2009cAsym', resolution=2, desc='DKT31', suffix='dseg'))
+        """Converts inputspec to files"""
+        if (
+            self.inputs.surface_target not in self._surfaces or
+            self.inputs.volume_target not in self._volumes
+        ):
+            raise NotImplementedError(
+                "Target space (surface: {0}, volume: {1}) is not supported".format(
+                    self.inputs.surface_target, self.inputs.volume_target
+                )
+            )
+
+        tpl_kwargs = {'suffix': 'dseg'}
+        annotation_files = None
+        if self.inputs.volume_target == "MNI152NLin2009cAsym":
+            tpl_kwargs.update({
+                'resolution': '2',
+                'desc': 'DKT31',
+            })
+            annotation_files = sorted(
+                glob(os.path.join(self.inputs.subjects_dir,
+                                  self.inputs.surface_target,
+                                  'label',
+                                  '*h.aparc.annot'))
+            )
+
+        elif self.inputs.volume_target == 'MNI152NLin6Asym':
+            res = '2' if self.inputs.density == '32k' else '6'
+
+            tpl_kwargs.update({
+                'atlas': 'HCP',
+                'resolution': res,
+            })
+            annotation_files = [
+                str(f) for f in get_template(
+                    'fsLR', density=self.inputs.density, desc='nomedialwall', suffix='dparc'
+                )
+            ]
+
+        if len(annotation_files) != 2:
+            raise IOError("Invalid number of surface annotation files")
+
+        label_file = str(get_template(self.inputs.volume_target, **tpl_kwargs))
         return annotation_files, label_file
 
     @staticmethod
-    def _create_cifti_image(bold_file, label_file, annotation_files, gii_files,
-                            volume_target, surface_target, tr):
+    def _create_cifti_image(bold_file, label_file, bold_surfs,
+                            targets, tr, annotation_files):
         """
         Generate CIFTI image in target space.
 
         Parameters
-        ----------
```

Comment:
```suggestion
        ----------
```

The section markers are needed for sphinxcontrib-napoleon to correctly format these docstrings